Extension/src/Debugger/configurationProvider.ts

```diff
@@ -188,85 +188,94 @@ class CppConfigurationProvider implements vscode.DebugConfigurationProvider {
 
-            // Validate LLDB-MI
-            if (os.platform() === 'darwin' && // Check for macOS
-                fs.existsSync(lldb_mi_10_x_path) && // lldb-mi 10.x exists
-                (!macOSMIMode || macOSMIMode === 'lldb') &&
-                !macOSMIDebuggerPath // User did not provide custom lldb-mi
-            ) {
-                const frameworkPath: string | undefined = this.getLLDBFrameworkPath();
+        // Validate LLDB-MI
+        if (os.platform() === 'darwin' && // Check for macOS
+            fs.existsSync(lldb_mi_10_x_path) && // lldb-mi 10.x exists
+            (!macOSMIMode || macOSMIMode === 'lldb') &&
+            !macOSMIDebuggerPath // User did not provide custom lldb-mi
+        ) {
+            const frameworkPath: string | undefined = this.getLLDBFrameworkPath();
 
-                if (!frameworkPath) {
-                    const moreInfoButton: string = localize("lldb.framework.install.xcode", "More Info");
-                    const LLDBFrameworkMissingMessage: string = localize("lldb.framework.not.found", "Unable to locate 'LLDB.framework' for lldb-mi. Please install XCode or XCode Command Line Tools.");
+            if (!frameworkPath) {
+                const moreInfoButton: string = localize("lldb.framework.install.xcode", "More Info");
+                const LLDBFrameworkMissingMessage: string = localize("lldb.framework.not.found", "Unable to locate 'LLDB.framework' for lldb-mi. Please install XCode or XCode Command Line Tools.");
 
-                    vscode.window.showErrorMessage(LLDBFrameworkMissingMessage, moreInfoButton)
-                        .then(value => {
-                            if (value === moreInfoButton) {
-                                const helpURL: string = "https://aka.ms/vscode-cpptools/LLDBFrameworkNotFound";
-                                vscode.env.openExternal(vscode.Uri.parse(helpURL));
-                            }
-                        });
+                vscode.window.showErrorMessage(LLDBFrameworkMissingMessage, moreInfoButton)
+                    .then(value => {
+                        if (value === moreInfoButton) {
+                            const helpURL: string = "https://aka.ms/vscode-cpptools/LLDBFrameworkNotFound";
+                            vscode.env.openExternal(vscode.Uri.parse(helpURL));
+                        }
+                    });
 
-                    return undefined;
-                }
+                return undefined;
             }
         }
-        // if config or type is not specified, return null to trigger VS Code to open a configuration file https://github.com/Microsoft/vscode/issues/54213
-        return config && config.type ? config : null;
+
+        if (config.logging?.engineLogging) {
+            const outputChannel: logger.Logger = logger.getOutputChannelLogger();
+            outputChannel.appendLine(JSON.stringify(config, undefined, 2));
```

Comment:
do you want to preface this with` Launch configuration` or something so it doesn't seem like you are just dumping a blob of json into the output window. Similar to what we do with the XML in MIEngine.